hyprlang-ts-mode.el

```diff
@@ -82,6 +82,7 @@
 (defvar hyprlang-ts-mode--indent-rules
   ;; Hyprlang indentation rules
   `((hyprlang
+     ((node-is "}") parent 0)
```

Comment:
I tried doing that initially, but unfortunately it doesn't work that way.  I suspect this is because the code that goes through these rules apply the first match, so if the `((parent-is "section") ...)` matcher is found first it will still indent brackets because the parent of that "node" is the section. (This is mainly speculation based on what I experienced during testing. I haven't found any documentation that explicitly says this.)

Moving the other two similar queries also wouldn't work because in that case comments and nested sections are not indented correctly.

A possible way to make this look cleaner would be to remove the three matchers after the parent check:
```
(defvar hyprlang-ts-mode--indent-rules
  ;; Hyprlang indentation rules
  `((hyprlang
     (no-node parent 0)
     ((node-is "}") parent 0)
     ((parent-is "section") parent hyprlang-ts-mode-indent-offset))))
```
Since these matchers were after `((parent-is "section") ...)` they only apply if the parent is not a section and those nodes are already not indented (and based on the grammar the only two valid parents for most nodes where indentation would be logical are the [section](https://github.com/tree-sitter-grammars/tree-sitter-hyprlang/blob/6858695eba0e63b9e0fceef081d291eb352abce8/src/grammar.json#L125) and the top level [configuration](https://github.com/tree-sitter-grammars/tree-sitter-hyprlang/blob/6858695eba0e63b9e0fceef081d291eb352abce8/src/grammar.json#L5)).

Reply:
Okay that makes sense. in this case lets merge this and we will work on the refactor later. the fix seems to work for me.